application/dash_application/plot_builder.py

```diff
@@ -75,19 +75,28 @@ def fill_in_shape_col(df: DataFrame, shape_col: str, shape_or_colour_values:
     return df
 
 def fill_in_colour_col(df: DataFrame, colour_col: str, shape_or_colour_values:
-        dict):
+        dict, highlight_samples=None):
     df['colour'] = pandas.Series
     if not df.empty:
         all_colours = get_colours_for_values(shape_or_colour_values[
                                             colour_col].tolist())
         # for each row,
         df['colour'] = df.apply(lambda row: all_colours.get(row[
             colour_col]), axis=1)
+
+    if highlight_samples:
+        df.loc[df[PINERY_COL.SampleName].isin(highlight_samples), 'colour'] = '#F00'
+    return df
+
+def fill_in_size_col(df: DataFrame):
```

Comment:
I'm a little leery of having this depend on the colour, as it'll introduce unexpected behaviour if this function happens to be called before `fill_in_colour_col`. Could this also make use of the `highlight_samples` parameter?

Reply:
Fixed